### PR TITLE
Fix image display for palette images with transparency

### PR DESCRIPTION
**Pull request by gandalf3**
Even if image is not in RGB or RGBA format, it could still contain
transparency information.

I used the "transparency" key in the image info dict, see documentation here: https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html
Basically this key only exists for some formats in some modes, when the image contains transparency. I didn't test any formats besides a rudimentary test of palette mode .png; this change may also affect .eps, .gif, .gd, .xpm.

#### ISSUE TYPE
- Bug fix

#### RUNTIME ENVIRONMENT

- Operating system and version:  Arch
- Terminal emulator and version: kitty 0.25.2
- Python version: 3.10.5
- Ranger version/commit: fcf44f6f12506fa0096cb6468de6a2706bbfaf51
- Locale: en_US.UTF-8

#### CHECKLIST

- [x] The `CONTRIBUTING` document has been read **[REQUIRED]**
- [ ] All changes follow the code style **[REQUIRED]**
- [ ] All new and existing tests pass **[REQUIRED]**
- [ ] Changes require config files to be updated
    - [ ] Config files have been updated
- [ ] Changes require documentation to be updated
    - [ ] Documentation has been updated
- [ ] Changes require tests to be updated
    - [ ] Tests have been updated

#### MOTIVATION AND CONTEXT

Fixes: https://github.com/ranger/ranger/issues/2690

#### TESTING

I ran the tests and got a 9.99/10 reported failure rate; but AFAICT there aren't any tests which cover this change.

### CODE REVIEW

**Comment by toonn** [comment]
Apologies for how long it's taken to get around to taking a look.

This seems like a pretty reasonable fix. I just pushed a commit which might simplify it further though.

The only case where my code will differ is if `info["transparency"] is None`, in which case my code converts to RGBA while the original converts to RGB. Can that key can ever map to `None` while a conversion to RGB is preferable?

Another simplification would be always converting to RGBA, if we can test a good amount of varied images in various formats and measure the difference in time and size of conversion to RGB/RGBA.

Let me know what you think?

**Comment by gandalf3** [comment]
No worries! Maintaining an insanely popular project has got to be really hard, yall are amazing! 

I actually found both style of checks within pillow itself, and while skimming that documentation I don't see anything about `None` being special. The no transparency case seems to be consistently represented by the absence of the `transparency` attribute in the image `info`, at least as described in the documentation.

So TL;DR your change looks good to me!

**Review by toonn — COMMENTED** [review]
LGTM

Thank you! Pretty good first contribution! 🎉 

I will try not to have it take so long to review and merge next time though.
